pkgs/test/coq/overrideCoqDerivation/default.nix

```diff
@@ -0,0 +1,40 @@
+{ lib, coq, mkCoqPackages, runCommandNoCC }:
+
+let
+
+  # This is just coq, but with dontFilter set to true. We need to set
+  # dontFilter to true here so that _all_ packages are visibile in coqPackages.
+  # There may be some versions of the top-level coq and coqPackages that don't
+  # build QuickChick, which is what we are using for this test below.
+  coqWithAllPackages = coq // { dontFilter = true; };
+
+  coqPackages = mkCoqPackages coqWithAllPackages;
+
+  # This is the main test.  This uses overrideCoqDerivation to
+  # override arguments to mkCoqDerivation.
+  #
+  # Here, we override the defaultVersion and release arguments to
+  # mkCoqDerivation.
+  overriddenQuickChick =
+    coqPackages.lib.overrideCoqDerivation
+      {
+        defaultVersion = "9999";
+        release."9999".sha256 = lib.fakeSha256;
+      }
+      coqPackages.QuickChick;
```

Comment:
Here's an example of using `overrideCoqDerivation`.  You can see that with this new function, end users would easily be able to override arguments to `mkCoqDerivation` that _aren't_ exposed from the Coq libraries themselves.

This `overrideCoqDerivation` function is very similar to the `overrideCabal` function from the Haskell package set.